src/whiteflag/wf_core/message.rs

```diff
@@ -0,0 +1,242 @@
+    pub subject_code: String,
+
+    pub reference_indicator: String,
+    pub referenced_message: String,
+
+    pub crypto_data_type: String,
+    pub crypto_data: String,
+
+    pub transaction_hash: String,
+    pub originator_address: String,
+
+    pub verification_method: String,
+    pub verification_data: String,
+}
+
+impl WhiteflagMessage {
+    pub fn new(message_code: String) -> WhiteflagMessage {
+        WhiteflagMessage {
+            prefix: "WF".to_string(),
+            version: "1".to_string(),
+            message_code: message_code.clone(),
+            message_type: message_code.clone(),
+            duress_indictor: "".to_string(),
+            encryption_indicator: "".to_string(),
+            object_type: "".to_string(),
+            subject_code: "".to_string(),
+            reference_indicator: "".to_string(),
+            referenced_message: "".to_string(),
+            crypto_data_type: "".to_string(),
+            crypto_data: "".to_string(),
+            transaction_hash: "".to_string(),
+            originator_address: "".to_string(),
+            verification_method: "".to_string(),
+            verification_data: "".to_string(),
+        }
+    }
+
+    pub fn is_valid(&self) -> bool {
+        self.message_code == self.message_type && self.prefix != "" && self.version == "1"
+    }
+
+    pub fn compile_auth_message(field_values: Vec<&str>) -> Option<WhiteflagMessage> {
+        let message = WhiteflagMessage {
+            prefix: field_values[0].to_string(),
+            version: field_values[1].to_string(),
+            message_code: field_values[4].to_string(),
+            message_type: "A".to_string(),
+            duress_indictor: field_values[3].to_string(),
+            encryption_indicator: field_values[2].to_string(),
+            object_type: "".to_string(),
+            subject_code: "".to_string(),
+            reference_indicator: field_values[5].to_string(),
+            referenced_message: field_values[6].to_string(),
+            crypto_data_type: "".to_string(),
+            crypto_data: "".to_string(),
+            transaction_hash: "".to_string(),
+            originator_address: "".to_string(),
+            verification_method: field_values[7].to_string(),
+            verification_data: field_values[8].to_string(),
+        };
+        if message.is_valid() {
+            return Some(message);
+        } else {
+            return None;
+        }
+    }
+
+    pub fn get_encryption_indicator(&self) -> String {
+        self.encryption_indicator.clone()
+    }
+    pub fn set_encryption_indicator(&mut self, arg: String) -> bool {
+        if self.encryption_indicator == "" {
+            self.encryption_indicator = arg;
+            return true;
+        }
+        return false;
+    }
+
+    pub fn get_subject_code(&self) -> String {
+        self.subject_code.clone()
+    }
+    pub fn set_subject_code(&mut self, arg: String) -> bool {
+        if self.subject_code == "" {
```

Comment:
would be good to make the fields that can be nothing `Option<>`'s for the rusty version

Reply:
correct -- there is a lot of work to do in this codebase to make it conform to idiomatic rust

I am not even sure we need all these `get()` and `set()` methods as they are more of a java convention